cpp/src/parquet/statistics.h

```diff
@@ -114,19 +115,79 @@ std::shared_ptr<TypedComparator<DType>> MakeComparator(const ColumnDescriptor* d
   return std::static_pointer_cast<TypedComparator<DType>>(Comparator::Make(descr));
 }
 
+class PARQUET_EXPORT EncodedGeometryStatistics {
+ public:
+  static constexpr double kInf = std::numeric_limits<double>::infinity();
+
+  EncodedGeometryStatistics() = default;
+  EncodedGeometryStatistics(const EncodedGeometryStatistics&) = default;
+  EncodedGeometryStatistics(EncodedGeometryStatistics&&) = default;
+  EncodedGeometryStatistics& operator=(const EncodedGeometryStatistics&) = default;
+
+  double xmin{kInf};
+  double xmax{-kInf};
+  double ymin{kInf};
+  double ymax{-kInf};
+  double zmin{kInf};
+  double zmax{-kInf};
+  double mmin{kInf};
+  double mmax{-kInf};
+  std::vector<std::pair<std::string, std::string>> coverings;
+  std::vector<uint32_t> geometry_types;
+
+  bool has_z() const { return (zmax - zmin) > 0; }
+
+  bool has_m() const { return (mmax - mmin) > 0; }
+};
+
+class GeometryStatisticsImpl;
+
+class PARQUET_EXPORT GeometryStatistics {
```

Comment:
IMO, `EncodedGeometryStatistics` is only used internally by the writer before converting to thrift statistics. We need to add access methods for bbox, covering and geometry_types to `GeometryStatistics` so that downstream users do not have to call `Encode()`.

Reply:
Several getters were added to `GeometryStatistics`; test case is also updated to use `GeometryStatistics` instead of `EncodedGeometryStatistics`.

Reply:
Thanks for adding theses!